### ISSUE TITLE
Selectbox item is not initially selected in strict mode using animations

### ISSUE THREAD

**Issue by marcospassos** [ariakit/ariakit#1564]
### Current behavior

Continuation of #1531.

After trying to figure out why the fix was working locally, I finally found it: the item still doesn't get focused when animated.

Here are a few observations that may help find out the root cause:

- Everything works as expected the second time the select is open
- Opening using the keyboard works fine, even on the first time

### Steps to reproduce the bug

1. Open sandbox: https://codesandbox.io/s/priceless-wave-ho98qc
2. Click on the select
3. Notice that the item isn't selected
4. Close the select and open it again
5. Notice that the first item is now selected

### Expected behavior

The first item gets selected when opening the selectbox.

### Workaround

https://github.com/ariakit/ariakit/issues/1564#issuecomment-1169985007

**Comment by marcospassos** [comment]
That's how it's, isn't it?
![image](https://user-images.githubusercontent.com/943036/176445015-81ac6cef-21f6-48ed-b48f-6bde56adfb01.png)

**Comment by diegohaz** [comment]
Did you save the file? 😄

**Comment by diegohaz** [comment]
For now, the workaround is to handle `data-enter`/`data-leave` yourself:

```jsx
  const { animated, contentElement, visible, animating } = select;
  const [transition, setTransition] = useState(null);

  useEffect(() => {
    if (!animated || !contentElement?.isConnected) {
      setTransition(null);
      return;
    }
    setTransition(visible ? "enter" : animating ? "leave" : null);
  }, [animated, contentElement, visible, animating]);

  <SelectPopover portal state={select} className="popover">
    {(props) => (
      <div
        {...props}
        data-enter={transition === "enter" ? "" : undefined}
        data-leave={transition === "leave" ? "" : undefined}
      >
       ...
      </div>
    )}
  </SelectPopover>
```

**Comment by diegohaz** [comment]
> 20% of the code is now workarounds 🥲, mainly because of this, #1501 and #1549.
> 
> I'm aware it's an unstable release. Still, we're using Arikit extensively in a large codebase to provide feedback and help improve the project, but all these workarounds are becoming kinda messy. I can help fix those issues with some guidance.
> 
> Do you already have an idea of how to approach these problems?

That's an essential part of the project's philosophy and one of the main reasons this library exists.

*As a developer*, I want to be able to fix bugs or implement new features in my project without having to convince dependency maintainers and wait until a solution is implemented upstream. Or, if I can contribute, I don't want to be stuck waiting until my PR is reviewed, approved, merged, and published, which can take a long time (and this tends to get "worse" as the project matures).

*As a maintainer*, I want to reduce my workload and mitigate the pressure of working on OSS knowing that users are stuck by a bug or feature that hasn't been implemented. This is one of the fundamental problems and leading causes of burnout and unsustainability in the OSS community.

When you tell me that 20% of your code is workaround, I understand that you managed to solve 20% of your problems without having to wait for RFCs and PRs written, reviewed, merged, published, etc.

That's why I'll always try to provide a workaround (and apply the `has workaround` label) to an issue before I (or another person) start solving it in the library codebase. You can always ignore the workaround and wait for the fix, review, release, etc., just like in any other library. It's just an option.

**Comment by diegohaz** [comment]
> Hey, @diegohaz, sorry if it sounded like a complaint – it wasn't my intention. I'm willing to contribute, as I said. I'm also an OSS maintainer, so I know how hard and stressful it is to maintain an OSS project sometimes. Please don't get me wrong, I am just offering help.

No worries. It didn't sound like a complaint. I just thought it was necessary to emphasize how important those workarounds are and that you can always choose not to implement them.

Working on Ariakit is generally awesome, and this methodology certainly helps a lot.

> But let's focus on the central part of the reply: _Do you already have an idea of how to approach these problems?_ Do you need help with something else?

Are you talking about the specific issues you linked? The suggested workarounds are my initial ideas.

The thing is that I can come up with them in a few minutes, but implementing the code in the library is not as easy. We may have to design APIs, write tests, test with screen readers and different browsers, submit PRs, review them, write changelogs, etc.

When people implement those suggestions in their real-world projects, we're kind of outsourcing part of this workload. They may report any issues and give feedback. I think this is a more efficient way to collaborate than asking everyone to engage in a PR without a concrete application. Sometimes, the final solution is totally different from the suggested workaround, but figuring this out takes time.

In other words, implementing the suggestions in your project and giving feedback as you've done on #1501 helps us find the best solution. That's the approach we've been following so far, and it's worked great.

Other things like #1549 are easier to solve. If you want to contribute to the codebase, you can check out the [contributing guide](https://github.com/ariakit/ariakit/blob/main/contributing.md).
